### PR TITLE
Setup docker for develop environment

### PR DESCRIPTION
**Pull request by quanpython**

### CODE REVIEW

**Comment by ReiHashimoto** [comment]
@quanpython 
cc: @reactplus-thom 

Thank you for your PR.
But I think hot reload isn't available with this configuraion.
We need to run build (for frontend) and restart fastapi server (for backend) on every time we change code.


FYI, in my case, 
I run backend(optinist) on port 8000 with `python main.py --reload`
and run frontend on port 3000 with `yarn start`
I use port 3000 to check frontend.


If you have any questions, please tell me!

File: frontend/Dockerfile.dev [hunk 1]
```diff
@@ -0,0 +1,6 @@
+FROM node:14.19.3-alpine
```
Comment (ReiHashimoto) [inline frontend/Dockerfile.dev:1]: Any reason for specifying node 14?

https://hub.docker.com/_/node/
node 14 seems to be out of supported tags.
Have your team developed on ver14?

Comment (reactplus-thom) [reply frontend/Dockerfile.dev:1]: @ReiHashimoto 
はい、FEソースはnode14で開発しているので、node14を指定しています。
※Node14では一番安定的に動きます。Node18にするとエラーができます。

File: Dockerfile.dev [hunk 2]
```diff
@@ -16,15 +16,18 @@ RUN mkdir -p /root/miniconda3 && \
...
+
+COPY optinist ./optinist
+
+COPY backend ./backend
+
+COPY frontend/build ./frontend/build
+
+COPY sample_data ./sample_data
+
+COPY main.py firebase_config.json firebase_private.json .env ./
```
Comment (ReiHashimoto) [inline Dockerfile.dev:31]: COPY required?
I think these files are already mounted on volumes in docker-compose.dev.yml

File: frontend/Dockerfile.dev (hunk 1 above)
Comment (ReiHashimoto) [reply frontend/Dockerfile.dev:1]: use v14 on this release

**Comment by ReiHashimoto** [comment]
@quanpython  cc: @reactplus-thom 

backend container exits with following exception.
Does this works correctly on your environment?

```
2023-06-22 16:58:25 Traceback (most recent call last):
2023-06-22 16:58:25   File "/app/backend/service/firebase/__init__.py", line 36, in <module>
2023-06-22 16:58:25     credential = credentials.Certificate('firebase_private.json')
2023-06-22 16:58:25   File "/usr/local/lib/python3.8/site-packages/firebase_admin/credentials.py", line 84, in __init__
2023-06-22 16:58:25     json_data = json.load(json_file)
2023-06-22 16:58:25   File "/usr/local/lib/python3.8/json/__init__.py", line 293, in load
2023-06-22 16:58:25     return loads(fp.read(),
2023-06-22 16:58:25   File "/usr/local/lib/python3.8/json/__init__.py", line 357, in loads
2023-06-22 16:58:25     return _default_decoder.decode(s)
2023-06-22 16:58:25   File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
2023-06-22 16:58:25     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
2023-06-22 16:58:25   File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
2023-06-22 16:58:25     raise JSONDecodeError("Expecting value", s, err.value) from None
2023-06-22 16:58:25 json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2023-06-22 16:58:25 
2023-06-22 16:58:25 During handling of the above exception, another exception occurred:
2023-06-22 16:58:25 
2023-06-22 16:58:25 Traceback (most recent call last):
2023-06-22 16:58:25   File "main.py", line 9, in <module>
2023-06-22 16:58:25     from optinist.__main_unit__ import main
2023-06-22 16:58:25   File "/app/optinist/__main_unit__.py", line 13, in <module>
2023-06-22 16:58:25     from backend.routers import auth, stat_image, user_manage
2023-06-22 16:58:25   File "/app/backend/routers/auth.py", line 3, in <module>
2023-06-22 16:58:25     from backend.deps import get_current_user
2023-06-22 16:58:25   File "/app/backend/deps.py", line 9, in <module>
2023-06-22 16:58:25     from backend.service.firebase.crud_user import read_user
2023-06-22 16:58:25   File "/app/backend/service/firebase/__init__.py", line 39, in <module>
2023-06-22 16:58:25     raise Exception("Error when initializing firebase-admin credential")
2023-06-22 16:58:25 Exception: Error when initializing firebase-admin credential
2023-06-22 17:00:27 Traceback (most recent call last):
2023-06-22 17:00:27   File "/app/backend/service/firebase/__init__.py", line 36, in <module>
2023-06-22 17:00:27     credential = credentials.Certificate('firebase_private.json')
2023-06-22 17:00:27   File "/usr/local/lib/python3.8/site-packages/firebase_admin/credentials.py", line 84, in __init__
2023-06-22 17:00:27     json_data = json.load(json_file)
2023-06-22 17:00:27   File "/usr/local/lib/python3.8/json/__init__.py", line 293, in load
2023-06-22 17:00:27     return loads(fp.read(),
2023-06-22 17:00:27   File "/usr/local/lib/python3.8/json/__init__.py", line 357, in loads
2023-06-22 17:00:27     return _default_decoder.decode(s)
2023-06-22 17:00:27   File "/usr/local/lib/python3.8/json/decoder.py", line 337, in decode
2023-06-22 17:00:27     obj, end = self.raw_decode(s, idx=_w(s, 0).end())
2023-06-22 17:00:27   File "/usr/local/lib/python3.8/json/decoder.py", line 355, in raw_decode
2023-06-22 17:00:27     raise JSONDecodeError("Expecting value", s, err.value) from None
2023-06-22 17:00:27 json.decoder.JSONDecodeError: Expecting value: line 1 column 1 (char 0)
2023-06-22 17:00:27 
2023-06-22 17:00:27 During handling of the above exception, another exception occurred:
2023-06-22 17:00:27 
2023-06-22 17:00:27 Traceback (most recent call last):
2023-06-22 17:00:27   File "main.py", line 9, in <module>
2023-06-22 17:00:27     from optinist.__main_unit__ import main
2023-06-22 17:00:27   File "/app/optinist/__main_unit__.py", line 13, in <module>
2023-06-22 17:00:27     from backend.routers import auth, stat_image, user_manage
2023-06-22 17:00:27   File "/app/backend/routers/auth.py", line 3, in <module>
2023-06-22 17:00:27     from backend.deps import get_current_user
2023-06-22 17:00:27   File "/app/backend/deps.py", line 9, in <module>
2023-06-22 17:00:27     from backend.service.firebase.crud_user import read_user
2023-06-22 17:00:27   File "/app/backend/service/firebase/__init__.py", line 39, in <module>
2023-06-22 17:00:27     raise Exception("Error when initializing firebase-admin credential")
2023-06-22 17:00:27 Exception: Error when initializing firebase-admin credential
```

**Review by itutu-tienday — APPROVED** [review]

**Pull request merged**